mock-plugin/src/test/java/io/mosip/signup/plugin/mock/service/MockProfileRegistryPluginImplTest.java

```diff
@@ -0,0 +1,272 @@
+        Mockito.when(restTemplate.exchange(
+                Mockito.anyString(),
+                Mockito.any(HttpMethod.class),
+                Mockito.any(),
+                Mockito.eq(new ParameterizedTypeReference<ResponseWrapper<MockIdentityResponse>>() {
+                }))).thenReturn(responseEntity);
+
+
+        ProfileResult profileResult = mockProfileRegistryPlugin.updateProfile(requestId, profileDto);
+        Assert.assertNotNull(profileResult);
+        Assert.assertEquals(profileResult.getStatus(),"UPDATED");
+    }
+
+    @Test
+    public void updateProfile_withOutVerifiedClaim_thenPass()  {
+
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "objectMapper",objectMapper);
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "identityEndpoint","http://localhost:8080/");
+        String requestId = "req-123";
+        String individualId = "ind-456";
+
+        Map<String, Object> verifiedData = new HashMap<>();
+        verifiedData.put("email","123@email.com");
+        verifiedData.put("password","123456");
+
+        JsonNode mockIdentity = objectMapper.valueToTree(verifiedData);
+        ProfileDto profileDto = new ProfileDto();
+        profileDto.setIndividualId(individualId);
+        profileDto.setIdentity(mockIdentity);
+
+
+        MockIdentityResponse mockIdentityResponse=new MockIdentityResponse();
+        mockIdentityResponse.setStatus("UPDATED");
+        ResponseWrapper<MockIdentityResponse> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setResponse(mockIdentityResponse);
+        ResponseEntity<ResponseWrapper<MockIdentityResponse>> responseEntity=new ResponseEntity<>(responseWrapper, HttpStatus.OK);
+        Mockito.when(restTemplate.exchange(
+                Mockito.anyString(),
+                Mockito.any(HttpMethod.class),
+                Mockito.any(),
+                Mockito.eq(new ParameterizedTypeReference<ResponseWrapper<MockIdentityResponse>>() {
+                }))).thenReturn(responseEntity);
+
+
+        ProfileResult profileResult = mockProfileRegistryPlugin.updateProfile(requestId, profileDto);
+        Assert.assertNotNull(profileResult);
+        Assert.assertEquals(profileResult.getStatus(),"UPDATED");
+    }
+
+    @Test
+    public void isMatch_withMatchingIdentityAndInputChallenge_thenPass() {
+        // Arrange
+        ObjectMapper objectMapper=new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "objectMapper",objectMapper);
+        Map<String, Object> verifiedData = new HashMap<>();
+        verifiedData.put("email","123@email.com");
+        verifiedData.put("password","123456");
+        verifiedData.put("UIN","1234567890");
+        JsonNode mockIdentity = objectMapper.valueToTree(verifiedData);
+
+        boolean isMatch = mockProfileRegistryPlugin.isMatch(mockIdentity, mockIdentity);
```

Comment:
you are passing the same variable both the arguments? what if internally they are doing just the object comparison, your test case will always pass. This is NOT correct.

Reply:
changed